ssl/tls13_enc.c

```diff
@@ -352,20 +367,23 @@ size_t tls13_final_finish_mac(SSL_CONNECTION *s, const char *str, size_t slen,
 int tls13_setup_key_block(SSL_CONNECTION *s)
 {
     const EVP_CIPHER *c;
+    const EVP_CIPHER *snc;
     const EVP_MD *hash;
     int mac_type = NID_undef;
     size_t mac_secret_size = 0;
 
     s->session->cipher = s->s3.tmp.new_cipher;
-    if (!ssl_cipher_get_evp(SSL_CONNECTION_GET_CTX(s), s->session, &c, &hash,
-                            &mac_type, &mac_secret_size, NULL, 0)) {
+    if (!ssl_cipher_get_evp(SSL_CONNECTION_GET_CTX(s), s->session, &snc, &c,
+                            &hash, &mac_type, &mac_secret_size, NULL, 0)) {
```

Comment:
Doesn't this also end up requiring a serial number encryption cipher for TLS 1.3 (not just DTLS)???

Reply:
Well spotted! Fixed.